hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1020,6 +1061,60 @@ public MembershipManagerCompat getMembershipManagerCompat() {
     }
 
     @Override
+    public void promoteLocalLiteMember() {
+        if (getClusterVersion().isLessThan(Versions.V3_9)) {
+            throw new UnsupportedOperationException("Lite member promotion is not available!");
+        }
+
+        MemberImpl member = getLocalMember();
+        if (!member.isLiteMember()) {
+            throw new IllegalStateException(member + " is not a lite member!");
+        }
+
+        MemberImpl master = getMasterMember();
+        PromoteLiteMemberOp op = new PromoteLiteMemberOp();
+        op.setCallerUuid(member.getUuid());
+
+        InternalCompletableFuture<MembersView> future =
+                nodeEngine.getOperationService().invokeOnTarget(SERVICE_NAME, op, master.getAddress());
+        MembersView view = future.join();
+
+        lock.lock();
+        try {
+            if (!member.getAddress().equals(master.getAddress())) {
```

Comment:
yes, possible. `MembershipManager#promoteToNormalMember()` applies the changes locally.